src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -100,7 +100,9 @@ public Pipe(PipeOptions options)
             _readerCompletion = default;
             _writerCompletion = default;
 
-            _pool = options.Pool;
+            // If we're using the default pool then mark it as null since we're just going to use the 
+            // array pool under the covers
+            _pool = options.Pool == MemoryPool<byte>.Shared ? null : options.Pool;
```

Comment:
If someone passes in  MemoryPool<byte>.Shared in the PipeOptions, we will still ignore it. That's fine?

Reply:
Yep.

Reply:
> If someone passes in  MemoryPool.Shared in the PipeOptions, we will still ignore it. That's fine?

Yes.  MemoryPool.Shared uses ArrayPool.Shared, just providing a thin veneer on top of it.  This change just avoids the additional overhead associated with MemoryPool.Shared, but with the same semantics.